src/main/java/ch/njol/skript/expressions/ExprItemCooldown.java

```diff
@@ -0,0 +1,142 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Item Cooldown")
+@Description("Set the cooldown of a specific material for a certain amount of ticks. Setting this to <code>0 ticks</code> will remove the cooldown.")
+@Examples({
+	"on right click using stick:" +
+	"\tset item cooldown of player's tool for player to 1 minute",
+	"\tset item cooldown of stone and grass for all players to 20 seconds",
+	"\treset item cooldown of cobblestone and dirt for all players"
+})
+@Since("INSERT VERSION")
+public class ExprItemCooldown extends SimpleExpression<Timespan> {
+	
+	static {
+		Skript.registerExpression(ExprItemCooldown.class, Timespan.class, ExpressionType.COMBINED, 
+			"[the] [item] cooldown of %itemtypes% for %players%",
+			"%players%'[s] [item] cooldown for %itemtypes%");
+	}
+	
+	private Expression<Player> players;
+	private Expression<ItemType> itemtypes;
+	
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		players = (Expression<Player>) exprs[1];
+		itemtypes = (Expression<ItemType>) exprs[0];
+		return true;
+	}
+	
+	@Override
+	protected Timespan[] get(Event e) {
+		Player[] players = this.players.getArray(e);
+		ItemType[] itemtypes = this.itemtypes.getArray(e);
+		if (players.length == 0 || itemtypes.length == 0)
+			return new Timespan[0];
+
+		Timespan[] timespan = new Timespan[players.length * itemtypes.length];
+		
+		int i = 0;
+		for (Player p : players) {
+			for (ItemType it : itemtypes) {
+				timespan[i] = Timespan.fromTicks_i(p.getCooldown(it.getMaterial()));
```

Comment:
getCooldown will throw an `IllegalArgumentException` if the material is not an item. A check to ensure it's not a block etc is required here. That will also impact the returning array, so maybe just an ArrayList or do the checking of the itemtypes to be items before creating the timespan array as you'll know the size after figuring out the itemtypes contents.